pkg/auth/auth_openshift.go

```diff
@@ -156,6 +158,18 @@ func (o *openShiftAuth) login(w http.ResponseWriter, token *oauth2.Token) (*logi
 func (o *openShiftAuth) logout(w http.ResponseWriter, r *http.Request) {
 	// NOTE: cookies are going away, this should be removed in the future
 
+	if r.Method != "POST" {
+		w.Header().Set("Allow", "POST")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Method unsupported, the only supported methods is POST"})
+		return
+	}
+
+	if err := VerifyCSRFToken(r); err != nil {
```

Comment:
We should also be verifying the source origin.

Is it possible instead to wrap logout in `authMiddleware`? It would be better not to repeat the logic here.